src/intervals/arithmetic.jl

```diff
@@ -302,7 +302,7 @@ end
 
 
 # Infimum and supremum of an interval
-inf(a::Interval) = a.lo
+inf(a::Interval) = ifelse(iszero(a.lo) && !signbit(a.lo), copysign(a.lo, -1), a.lo)
 sup(a::Interval) = a.hi
```

Comment:
should a similar check also be here for `sup` I am mainly thinking the case `sup(interval(0.0, -0.0))`. Maybe something like
```julia
sup(a::Interval) = ifelse(iszero(a.hi), abs(a.hi), a.hi)
```

or using the copysign and signbit if it's more performant.

Reply:
I don't think anything is needed with regards to `sup`, unless there is some specification in the standard that we haven't considered yet. 

As shown [above](https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/500#issuecomment-993103088), an interval `a = -0 .. -0` is such that `a.lo=0.0` and `a.hi=0.0`, so the signbit is already taken care of. That's the reason `sup(a)` is simply `a.hi`.

Reply:
yes I think if you move the normalisation before the `if validity_check` then it would work. I was mainly thinking to the example [below](https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/500#discussion_r768718450)